app/controllers/users_controller.rb

```diff
@@ -24,4 +28,10 @@ def update
   def user_params
     params.require(:user).permit(:first_name, :last_name, :email, :state, :user_name, :profile_photo)
   end
+
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      raise ActionController::RoutingError.new("Not Found")
```

Comment:
This is always an interesting choice; whether to display a page saying "you are not authorized to view this content, please sign in" or to display a 404 and pretend the page doesn't even exist. No wrong decisions, just be aware that you are making a choice.